main.tf

```diff
@@ -84,9 +84,9 @@ data "aws_iam_policy_document" "default" {
 }
 
 module "dns" {
-  source          = "git::https://github.com/cloudposse/tf_vanity?ref=tags/0.2.0"
-  aliases         = "${var.dns_zone_id ? list(var.hostname) : list()}"
-  zone_id         = "${var.dns_zone_id}"
-  target_dns_name = "${aws_s3_bucket.default.bucket_domain_name}"
+  source          = "git::https://github.com/cloudposse/tf_vanity.git?ref=tags/0.2.0"
+  aliases         = "${list(var.hostname)}"
```

Comment:
can you try instead `compact(list(var.dns_zone_id ? var.hostname : "")))`

Reply:
in this case get error:

```sh
* module.dns.var.aliases: __builtin_StringToBool: strconv.ParseBool: parsing "Z21GZDYDBU741J": invalid syntax in:

${compact(list(var.dns_zone_id ? var.hostname : ""))}
```

Reply:
I don't think that make sense because `parent_zone_id` is required in `tf_vanity`

Reply:
The objective is to make it *optional*. Since `count` is not a valid attribute of modules, we pass it along to the underlying resources instead

Reply:
By making it optional, DNS registration is not required.

Reply:
Okay, I should have written something more like:
```
${compact(list(length(var.parent_zone_id) + length(var.parent_zone_name) >0 ? var.hostname : ""))}
```

Reply:
something like `compact(list(length(var.dns_zone_id) > 0 ? var.hostname : ""))` can work